### DISCUSSION TITLE
Question: Making the plugin more general

### DISCUSSION THREAD

**Discussion by Menrath** [rugk/mastodon-simplified-federation#124] [Ideas]
Could it be possible to make this plugin even more general: So that it might also pickup any "reply to that fediverse post via my mastodon" button?

**Comment by rugk** [comment]
What do you mean exactly? You mean other Fediverse instances/software?

Yeah some functionality is already included actually. See https://github.com/rugk/mastodon-simplified-federation/tree/main/src/background/modules/Detect for the detection modules.

That said, the feature state is not that big and relies on contributions. And there is certainly more testing needed, see also https://github.com/rugk/mastodon-simplified-federation/issues/45

**Reply by Menrath** [reply]
Yes! In my case: I currently work on improving the ActivityPub plugin for WordPress. Making the reply button there have the same functions like on other Mastodon instances, that would be nice. 

Thanks for giving the links as a starting point. Something that is not clear on first sight: are the modules within `Detect` for software instances I click the reply button on, or for the target software I get redirected to?

**Reply by Menrath** [reply]
@rugk Actually I was wondering if there would be something like a special HTML attribute I could set within WordPress on the reply buttons that makes the detection for the plugin easy. That would also be the only way, as WordPress is highly customized. I mean when I contribute to this plugin I would need it anyway, but if a well known identifier for fediverse reply buttons already exist I would use it. That could also lead to a FEP https://codeberg.org/fediverse/fep.

**Reply by rugk** [reply]
Target.

Also it's only the redirection that catches it. If you have html that is more complicated.

There is nothing generic you can set as of now.
